<a href="https://colab.research.google.com/github/shubhpjoshi/EDA_Airbnb_project/blob/main/SJ_%7C_EDA_project_airbnb_booking_analysis.ipynb" target="_parent"><img src="https://colab.research.google.com/assets/colab-badge.svg" alt="Open In Colab"/></a>

##**Project Name**   - Airbnb booking analysis

##### **Project Type**    - EDA
##### **Contribution**    - Individual
##### **Team Member**       - Shubham Joshi

# **Project Summary -**

In the project we are analyzing Airbnb’s New York City(NYC) data of 2019. NYC is not only the famous city in the world but also has top global destination for visitors attracted to its museums, entertainment, restaurants, UN offices and commerce.

The project began with a comprehensive understanding of the Airbnb dataset, including data size, information like properties and their availability, price, location, reviews and ratings, exploring data related to Airbnb listings, including the number of properties listed, host characteristics, the variety of amenities available, and the occupancy rate of different properties etc. Further analysis of data to understand the significance of the reviews left by Airbnb users.

Exploratory data analysis projects on Airbnb typically involve investigating patterns and trends in various aspects of the platform, such as pricing, popularity and availability of listings.

This data can be used to gain insights into consumer behavior and preferences, as well as to inform marketing and business strategies for hosts and Airbnb as a company.

Techniques such as data visualization and objective solution may be used to analyze the data and draw meaningful conclusions.

In this type of analysis, data visualizations such as line plots, scatter plots, and bar charts are used to help identify trends, patterns, and relationships in the data. For instance, a bar chart can be used to show the distribution of properties across different neighbourhoods in a city.

**Overall**, the exploratory data analysis provides crucial insights for the Airbnb platform to improve customer satisfaction and enhance rental revenues. The insights also benefitted renters who can use the data generated to gain a deeper understanding of the landscape and make informed decisions.

# **Objectives of project:**
* **Comprehensive Understanding of Airbnb Dataset**: The project aims to thoroughly understand the Airbnb dataset for New York City in 2019. This includes examining various aspects such as property information, availability, pricing, location, reviews, and ratings.
* **Exploration of Airbnb Listings**: The project involves exploring data related to Airbnb listings, including the number of properties listed, host characteristics, available amenities, and occupancy rates of different properties.
* **Analysis of User Reviews**: The significance of reviews left by Airbnb users is analyzed to better understand their impact on the platform.
* **Investigation of Patterns and Trends**: Exploratory data analysis is conducted to investigate patterns and trends in various aspects of Airbnb, including pricing, popularity, and listing availability.

* **Insights into Consumer Behavior**: The project seeks to gain insights into consumer behavior and preferences based on Airbnb data. These insights can inform marketing and business strategies for hosts and Airbnb as a company.

### **Github Link  -**

# **1. Know your dataset**

In [2]:
# Importing Libraries

import numpy as np       # for computationally efficient operations
import pandas as pd      # for data manipulation, aggregation
import matplotlib.pyplot as plt
%matplotlib inline
import seaborn as sns     # For visulizations
from datetime import datetime

In [13]:
# Loading Dataset

file_path = '/content/Airbnb NYC 2019 (2).csv'
airbnb_df = pd.read_csv(file_path)

In [14]:
# Dataset first look

airbnb_df

id                                               name   host_id  \
0          2539                 Clean & quiet apt home by the park      2787   
1          2595                              Skylit Midtown Castle      2845   
2          3647                THE VILLAGE OF HARLEM....NEW YORK !      4632   
3          3831                    Cozy Entire Floor of Brownstone      4869   
4          5022   Entire Apt: Spacious Studio/Loft by central park      7192   
...         ...                                                ...       ...   
48890  36484665    Charming one bedroom - newly renovated rowhouse   8232441   
48891  36485057      Affordable room in Bushwick/East Williamsburg   6570630   
48892  36485431            Sunny Studio at Historical Neighborhood  23492952   
48893  36485609               43rd St. Time Square-cozy single bed  30985759   
48894  36487245  Trendy duplex in the very heart of Hell's Kitchen  68119814   

           host_name neighbourhood_group       neighbourhood  latitude  \
0               John            Brooklyn          Kensington  40.64749   
1           Jennifer           Manhattan             Midtown  40.75362   
2          Elisabeth           Manhattan              Harlem  40.80902   
3        LisaRoxanne            Brooklyn        Clinton Hill  40.68514   
4              Laura           Manhattan         East Harlem  40.79851   
...              ...                 ...                 ...       ...   
48890        Sabrina            Brooklyn  Bedford-Stuyvesant  40.67853   
48891        Marisol            Brooklyn            Bushwick  40.70184   
48892  Ilgar & Aysel           Manhattan              Harlem  40.81475   
48893            Taz           Manhattan      Hell's Kitchen  40.75751   
48894     Christophe           Manhattan      Hell's Kitchen  40.76404   

       longitude        room_type  price  minimum_nights  number_of_reviews  \
0      -73.97237     Private room    149               1                  9   
1      -73.98377  Entire home/apt    225               1                 45   
2      -73.94190     Private room    150               3                  0   
3      -73.95976  Entire home/apt     89               1                270   
4      -73.94399  Entire home/apt     80              10                  9   
...          ...              ...    ...             ...                ...   
48890  -73.94995     Private room     70               2                  0   
48891  -73.93317     Private room     40               4                  0   
48892  -73.94867  Entire home/apt    115              10                  0   
48893  -73.99112      Shared room     55               1                  0   
48894  -73.98933     Private room     90               7                  0   

      last_review  reviews_per_month  calculated_host_listings_count  \
0      2018-10-19               0.21                               6   
1      2019-05-21               0.38                               2   
2             NaN                NaN                               1   
3      2019-07-05               4.64                               1   
4      2018-11-19               0.10                               1   
...           ...                ...                             ...   
48890         NaN                NaN                               2   
48891         NaN                NaN                               2   
48892         NaN                NaN                               1   
48893         NaN                NaN                               6   
48894         NaN                NaN                               1   

       availability_365  
0                   365  
1                   355  
2                   365  
3                   194  
4                     0  
...                 ...  
48890                 9  
48891                36  
48892                27  
48893                 2  
48894                23  

[48895 rows x 16 columns]

In [15]:
# Shape of the dataset
airbnb_df.shape

(48895, 16)

In [16]:
# Basic information about the data
airbnb_df.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 48895 entries, 0 to 48894
Data columns (total 16 columns):
 #   Column                          Non-Null Count  Dtype  
---  ------                          --------------  -----  
 0   id                              48895 non-null  int64  
 1   name                            48879 non-null  object 
 2   host_id                         48895 non-null  int64  
 3   host_name                       48874 non-null  object 
 4   neighbourhood_group             48895 non-null  object 
 5   neighbourhood                   48895 non-null  object 
 6   latitude                        48895 non-null  float64
 7   longitude                       48895 non-null  float64
 8   room_type                       48895 non-null  object 
 9   price                           48895 non-null  int64  
 10  minimum_nights                  48895 non-null  int64  
 11  number_of_reviews               48895 non-null  int64  
 12  last_review                     

#### Duplicates values

In [19]:
# Checking Duplicate values
airbnb_df.duplicated().sum()
airbnb_df.drop_duplicates(inplace = True)

#### Missing values/null values

In [20]:
# Chaecking null values/Missing Values
missing_vaules = airbnb_df.isnull().sum()  # Only four column has null values
missing_vaules

id                                    0
name                                 16
host_id                               0
host_name                            21
neighbourhood_group                   0
neighbourhood                         0
latitude                              0
longitude                             0
room_type                             0
price                                 0
minimum_nights                        0
number_of_reviews                     0
last_review                       10052
reviews_per_month                 10052
calculated_host_listings_count        0
availability_365                      0
dtype: int64

The missing values table reveals that four columns contain null values: 'name' and 'host_name' (although similar entries with different IDs may exist), as well as 'last_reviews' and 'reviews_per_month' (which may be null for listings that have never received a review, although 'reviews_per_month' has data implications).

# What did you know about your dataset?


We can see our dataset contains 48,895 rows/entries and 16 columns/attributes. Let's delve into the details of these variables:

* id: A unique identifier for an Airbnb listing.
* name: The title or name of the listed property or room type on the platform.
*host_id: A unique identifier for an Airbnb host.
*host_name: The name of the person under whom the host is registered.
*neighbourhood_group: A grouping of areas.
*neighbourhood: The specific area falling under the neighbourhood_group.
*latitude: The latitude coordinate of the listing.
*longitude: The longitude coordinate of the listing.
*room_type: A categorization of the type of listing rooms.
*price: The price of the listing.
*minimum_nights: The minimum number of nights required for a single visit.
*number_of_reviews: The total count of reviews provided by visitors.
*last_review: The content of the last review given.
*reviews_per_month: An indicator of reviews received per month, highlighting the activity of the listing.
*calculated_host_listings_count: The total number of listings registered under the host.
*availability_365: The number of days in a year for which a host makes their property available.







## 2. Understanding variables

In [21]:
# Dataset columns
airbnb_df.columns

Index(['id', 'name', 'host_id', 'host_name', 'neighbourhood_group',
       'neighbourhood', 'latitude', 'longitude', 'room_type', 'price',
       'minimum_nights', 'number_of_reviews', 'last_review',
       'reviews_per_month', 'calculated_host_listings_count',
       'availability_365'],
      dtype='object')

In [22]:
#  Describing dataset with continuous variables
# We are excluding latitude & longitude as they are coordinates and id & host_id as they're unique, on unique values statistical operations would not give desired insight.
airbnb_df[['availability_365',	'calculated_host_listings_count',	'minimum_nights', 'number_of_reviews',	'price','reviews_per_month']].describe()

availability_365  calculated_host_listings_count  minimum_nights  \
count      48895.000000                    48895.000000    48895.000000   
mean         112.781327                        7.143982        7.029962   
std          131.622289                       32.952519       20.510550   
min            0.000000                        1.000000        1.000000   
25%            0.000000                        1.000000        1.000000   
50%           45.000000                        1.000000        3.000000   
75%          227.000000                        2.000000        5.000000   
max          365.000000                      327.000000     1250.000000   

       number_of_reviews         price  reviews_per_month  
count       48895.000000  48895.000000       38843.000000  
mean           23.274466    152.720687           1.373221  
std            44.550582    240.154170           1.680442  
min             0.000000      0.000000           0.010000  
25%             1.000000     69.000000           0.190000  
50%             5.000000    106.000000           0.720000  
75%            24.000000    175.000000           2.020000  
max           629.000000  10000.000000          58.500000

### Variables Description

So, we get to know that some columns falls under categorical and remaining are numeriacal except one last_review comes under Date_Time category.
*   Categorical variable : name, host_name, neighbourhood_group,
       neighbourhood, room_type, price
*   Numerical variable : id, host_id, price,
       minimum_nights, number_of_reviews, reviews_per_month, calculated_host_listings_count, availability_365
*   Date_Time variable : last_review
*  Coordinates : latitude, longitude

### checking the unique values

In [23]:
(airbnb_df['neighbourhood_group'].value_counts()) #area of the city

Manhattan        21661
Brooklyn         20104
Queens            5666
Bronx             1091
Staten Island      373
Name: neighbourhood_group, dtype: int64

In [24]:
price_value_counts = airbnb_df['price'].value_counts()
price_value_counts.sort_values(ascending = False)    # There are 673 unique prices in the dataset, spanning from 0 to 10,000.

100     2051
150     2047
50      1534
60      1458
200     1401
        ... 
1497       1
2850       1
990        1
8000       1
338        1
Name: price, Length: 674, dtype: int64

In [25]:
airbnb_df['neighbourhood'].value_counts()   # no. of the neighbourhood

Williamsburg          3920
Bedford-Stuyvesant    3714
Harlem                2658
Bushwick              2465
Upper West Side       1971
                      ... 
Fort Wadsworth           1
Richmondtown             1
New Dorp                 1
Rossville                1
Willowbrook              1
Name: neighbourhood, Length: 221, dtype: int64

In [26]:
airbnb_df['host_name'].nunique()    #11.5k hosts and 49k listings shows single host have multiple listings.

11452

In [27]:
airbnb_df['last_review'].nunique()

1764

In [28]:
airbnb_df['minimum_nights'].nunique()

109

In [29]:
airbnb_df['calculated_host_listings_count'].unique()   # counts of unique host

array([  6,   2,   1,   4,   3,   5,   7,  13,  28,  11,   8,   9,  52,
        18,  15,  19,  10,  39,  26,  29,  12,  21,  96,  14,  34,  43,
       121,  37,  49,  31,  91,  16,  87,  33,  23,  50,  20,  25, 232,
        17,  47, 103,  65,  30,  27, 327,  32])

In [30]:
# Check Unique Values for each variable.
#looks all the property ids are different and each listings are different here!
airbnb_df['id'].nunique()

48895

In [31]:
airbnb_df['name'].nunique() #shows some listing names are common

47905

In [32]:
airbnb_df['name'].nunique()  #shows some listing names are common

47905

# **3. DATA WRANGLING**

In [33]:
airbnb_df.drop(['id','last_review'],axis = 1,inplace =True)   # dropping unnecessary columns

In [34]:
# Examining changes after droping unnecessary columns
airbnb_df.shape

(48895, 14)

In [35]:
airbnb_df

name   host_id  \
0                     Clean & quiet apt home by the park      2787   
1                                  Skylit Midtown Castle      2845   
2                    THE VILLAGE OF HARLEM....NEW YORK !      4632   
3                        Cozy Entire Floor of Brownstone      4869   
4       Entire Apt: Spacious Studio/Loft by central park      7192   
...                                                  ...       ...   
48890    Charming one bedroom - newly renovated rowhouse   8232441   
48891      Affordable room in Bushwick/East Williamsburg   6570630   
48892            Sunny Studio at Historical Neighborhood  23492952   
48893               43rd St. Time Square-cozy single bed  30985759   
48894  Trendy duplex in the very heart of Hell's Kitchen  68119814   

           host_name neighbourhood_group       neighbourhood  latitude  \
0               John            Brooklyn          Kensington  40.64749   
1           Jennifer           Manhattan             Midtown  40.75362   
2          Elisabeth           Manhattan              Harlem  40.80902   
3        LisaRoxanne            Brooklyn        Clinton Hill  40.68514   
4              Laura           Manhattan         East Harlem  40.79851   
...              ...                 ...                 ...       ...   
48890        Sabrina            Brooklyn  Bedford-Stuyvesant  40.67853   
48891        Marisol            Brooklyn            Bushwick  40.70184   
48892  Ilgar & Aysel           Manhattan              Harlem  40.81475   
48893            Taz           Manhattan      Hell's Kitchen  40.75751   
48894     Christophe           Manhattan      Hell's Kitchen  40.76404   

       longitude        room_type  price  minimum_nights  number_of_reviews  \
0      -73.97237     Private room    149               1                  9   
1      -73.98377  Entire home/apt    225               1                 45   
2      -73.94190     Private room    150               3                  0   
3      -73.95976  Entire home/apt     89               1                270   
4      -73.94399  Entire home/apt     80              10                  9   
...          ...              ...    ...             ...                ...   
48890  -73.94995     Private room     70               2                  0   
48891  -73.93317     Private room     40               4                  0   
48892  -73.94867  Entire home/apt    115              10                  0   
48893  -73.99112      Shared room     55               1                  0   
48894  -73.98933     Private room     90               7                  0   

       reviews_per_month  calculated_host_listings_count  availability_365  
0                   0.21                               6               365  
1                   0.38                               2               355  
2                    NaN                               1               365  
3                   4.64                               1               194  
4                   0.10                               1                 0  
...                  ...                             ...               ...  
48890                NaN                               2                 9  
48891                NaN                               2                36  
48892                NaN                               1                27  
48893                NaN                               6                 2  
48894                NaN                               1                23  

[48895 rows x 14 columns]

In [38]:
# We know that reviews_per_month column have many null values, we will replace it with '0'.
# And name and host_name also have some empty indexes, replace it with 'Unknown' and 'no_name' resp.
replacement_values = {'reviews_per_month': 0, 'host_name': 'no_name', 'name': 'Unknown'}
airbnb_df.fillna(replacement_values, inplace = True)
airbnb_df

name   host_id  \
0                     Clean & quiet apt home by the park      2787   
1                                  Skylit Midtown Castle      2845   
2                    THE VILLAGE OF HARLEM....NEW YORK !      4632   
3                        Cozy Entire Floor of Brownstone      4869   
4       Entire Apt: Spacious Studio/Loft by central park      7192   
...                                                  ...       ...   
48890    Charming one bedroom - newly renovated rowhouse   8232441   
48891      Affordable room in Bushwick/East Williamsburg   6570630   
48892            Sunny Studio at Historical Neighborhood  23492952   
48893               43rd St. Time Square-cozy single bed  30985759   
48894  Trendy duplex in the very heart of Hell's Kitchen  68119814   

           host_name neighbourhood_group       neighbourhood  latitude  \
0               John            Brooklyn          Kensington  40.64749   
1           Jennifer           Manhattan             Midtown  40.75362   
2          Elisabeth           Manhattan              Harlem  40.80902   
3        LisaRoxanne            Brooklyn        Clinton Hill  40.68514   
4              Laura           Manhattan         East Harlem  40.79851   
...              ...                 ...                 ...       ...   
48890        Sabrina            Brooklyn  Bedford-Stuyvesant  40.67853   
48891        Marisol            Brooklyn            Bushwick  40.70184   
48892  Ilgar & Aysel           Manhattan              Harlem  40.81475   
48893            Taz           Manhattan      Hell's Kitchen  40.75751   
48894     Christophe           Manhattan      Hell's Kitchen  40.76404   

       longitude        room_type  price  minimum_nights  number_of_reviews  \
0      -73.97237     Private room    149               1                  9   
1      -73.98377  Entire home/apt    225               1                 45   
2      -73.94190     Private room    150               3                  0   
3      -73.95976  Entire home/apt     89               1                270   
4      -73.94399  Entire home/apt     80              10                  9   
...          ...              ...    ...             ...                ...   
48890  -73.94995     Private room     70               2                  0   
48891  -73.93317     Private room     40               4                  0   
48892  -73.94867  Entire home/apt    115              10                  0   
48893  -73.99112      Shared room     55               1                  0   
48894  -73.98933     Private room     90               7                  0   

       reviews_per_month  calculated_host_listings_count  availability_365  
0                   0.21                               6               365  
1                   0.38                               2               355  
2                   0.00                               1               365  
3                   4.64                               1               194  
4                   0.10                               1                 0  
...                  ...                             ...               ...  
48890               0.00                               2                 9  
48891               0.00                               2                36  
48892               0.00                               1                27  
48893               0.00                               6                 2  
48894               0.00                               1                23  

[48895 rows x 14 columns]

In [41]:
# again examining changes
airbnb_df.host_name.isnull().sum()

0

In [42]:
airbnb_df

name   host_id  \
0                     Clean & quiet apt home by the park      2787   
1                                  Skylit Midtown Castle      2845   
2                    THE VILLAGE OF HARLEM....NEW YORK !      4632   
3                        Cozy Entire Floor of Brownstone      4869   
4       Entire Apt: Spacious Studio/Loft by central park      7192   
...                                                  ...       ...   
48890    Charming one bedroom - newly renovated rowhouse   8232441   
48891      Affordable room in Bushwick/East Williamsburg   6570630   
48892            Sunny Studio at Historical Neighborhood  23492952   
48893               43rd St. Time Square-cozy single bed  30985759   
48894  Trendy duplex in the very heart of Hell's Kitchen  68119814   

           host_name neighbourhood_group       neighbourhood  latitude  \
0               John            Brooklyn          Kensington  40.64749   
1           Jennifer           Manhattan             Midtown  40.75362   
2          Elisabeth           Manhattan              Harlem  40.80902   
3        LisaRoxanne            Brooklyn        Clinton Hill  40.68514   
4              Laura           Manhattan         East Harlem  40.79851   
...              ...                 ...                 ...       ...   
48890        Sabrina            Brooklyn  Bedford-Stuyvesant  40.67853   
48891        Marisol            Brooklyn            Bushwick  40.70184   
48892  Ilgar & Aysel           Manhattan              Harlem  40.81475   
48893            Taz           Manhattan      Hell's Kitchen  40.75751   
48894     Christophe           Manhattan      Hell's Kitchen  40.76404   

       longitude        room_type  price  minimum_nights  number_of_reviews  \
0      -73.97237     Private room    149               1                  9   
1      -73.98377  Entire home/apt    225               1                 45   
2      -73.94190     Private room    150               3                  0   
3      -73.95976  Entire home/apt     89               1                270   
4      -73.94399  Entire home/apt     80              10                  9   
...          ...              ...    ...             ...                ...   
48890  -73.94995     Private room     70               2                  0   
48891  -73.93317     Private room     40               4                  0   
48892  -73.94867  Entire home/apt    115              10                  0   
48893  -73.99112      Shared room     55               1                  0   
48894  -73.98933     Private room     90               7                  0   

       reviews_per_month  calculated_host_listings_count  availability_365  
0                   0.21                               6               365  
1                   0.38                               2               355  
2                   0.00                               1               365  
3                   4.64                               1               194  
4                   0.10                               1                 0  
...                  ...                             ...               ...  
48890               0.00                               2                 9  
48891               0.00                               2                36  
48892               0.00                               1                27  
48893               0.00                               6                 2  
48894               0.00                               1                23  

[48895 rows x 14 columns]

In [46]:
# Inspecting properties that remain occupied throughout the year.
busy_properties = (airbnb_df['availability_365'] == 0).sum()
busy_properties

17533

In [ ]:
properties_with_zero_price = airbnb_df[airbnb_df['price'] == 0] #some properties have $0 listing price
properties_with_zero_price

In [55]:
properties_with_zero_price.shape[0] #the fare less than $10 doesn't have listings other than $0 Price.

11

In [58]:
# Remove the 0 listing price which dose not make sence
airbnb_df.drop(airbnb_df[airbnb_df['price'] == 0].index, inplace=True)

In [59]:
airbnb_df.info()  #updated data after dropping zero price rows and filling null values

<class 'pandas.core.frame.DataFrame'>
Int64Index: 48884 entries, 0 to 48894
Data columns (total 14 columns):
 #   Column                          Non-Null Count  Dtype  
---  ------                          --------------  -----  
 0   name                            48884 non-null  object 
 1   host_id                         48884 non-null  int64  
 2   host_name                       48884 non-null  object 
 3   neighbourhood_group             48884 non-null  object 
 4   neighbourhood                   48884 non-null  object 
 5   latitude                        48884 non-null  float64
 6   longitude                       48884 non-null  float64
 7   room_type                       48884 non-null  object 
 8   price                           48884 non-null  int64  
 9   minimum_nights                  48884 non-null  int64  
 10  number_of_reviews               48884 non-null  int64  
 11  reviews_per_month               48884 non-null  float64
 12  calculated_host_listings_count  

### What all manipulations have you done and insights you found?

**From the above experiments we get some more insights like :**
*   Entire home/apt have highest listings followed by Private room and Shared room have miniscule conteibution.
*   Unique host_name number and above experiment shows that one host have many roomtype and/or more than one listing in same and/or different neighbourhood.
*   Overall listing has distibuted in 5 neighbourhood_group which are having over 200 neighbourhoods. Its price goes upto 10k.
*   After inspection I figured out that a particular property name have one particular host_name hosted by that same individual but a particular host_name can have multiple properties in a neighbourhood_group or neighbourhood.
*   From the unique ids we get to know that all the property ids are different and each listings are different here.
*   By experimenting we get to know that in columns "price" and "availability_365" shows zero cost and not available throughout year respectively. Hosts not available round the justifies but zero Price doesn't.